File: csharpcore/GildedRose/Models/BackStagePasses.cs
using GildedRose.Abstraction;
using GildedRoseKata;

namespace GildedRose.Models
{
    public class BackStagePasses : Item, ICustomMethod
    {
        public int SellDaysGone { get; set; }

        public void UpdateQuality()
        {
            if (this.SellDaysGone > this.SellIn)
            {
                this.Quality = 0;
                return;
            }

            if ((this.SellIn - this.SellDaysGone) <= 10)
            {
                if (this.Quality < 51)
                    this.Quality += 2;
                return;
            }

            if ((this.SellIn - this.SellDaysGone) <= 5)
            {
                if (this.Quality < 51)
                    this.Quality += 3;
                return;
            }
        }

        public void UpdateSellIn()
        {
            if (this.SellIn > 0)
                this.SellIn--;
        }
    }
}

File: python/src/backstage_passes.py
from python.components.gilded_rose import GildedRose, Item


class BackstagePassesLogic(GildedRose):
    def __init__(self, name, quality, sell_in):
        super().__init__(Item(name, quality, sell_in))

    def update_quality(self):
        if self.quality < 50:
            self.quality += 1
        if self.sell_in < 11 and self.quality < 50:
            self.quality += 1
        if self.sell_in < 6 and self.quality < 50:
            self.quality += 1
        self.sell_in -= 1
        if self.sell_in < 0:
            self.quality = 0

        return self.quality, self.sell_in